### PR TITLE
[MENFORCER-371] Require maven 3.1.1

### PR DESCRIPTION
**Pull request by KroArtem**
I know this MR is not formed in a correct way. Just want to be sure these changes are acceptable for the plugin.

Based on https://github.com/apache/maven-clean-plugin/commit/114937a87a8f48c30fc2acd20ed707dbd90ab4a7 I thought maven version should be aligned here as well. I had to add `aether-api` to dependencies since it's used in EnforcerTestUtils.java but I believe its usage is not needed (cast `null` to `RepositorySystemSession`)

Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MENFORCER) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MENFORCER-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MENFORCER-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [ ] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [ ] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Review by slachiewicz — APPROVED** [review]

**Comment by KroArtem** [comment]
@slachiewicz , please check the description at the beginning :)

If everything is ok I'll create a corresponding JIRA ticket and will change the commit message.

**Comment by slachiewicz** [comment]
Thank you.

Yes, please create ticket. I was trying to remove more dependency to older Maven version, I'll rebase my work on top of your changes

**Comment by KroArtem** [comment]
@slachiewicz , done. I'm still not happy with declaring direct dependency to aether-api, though.

**Comment by slachiewicz** [comment]
I have a feeling that would be good to clean up all dependencies also - to avoid any nasty exclusions or overwrites.
I started with maven-common-artifact-filters, maven-dependency-tree and maven-dependency-analyzer. 
Let's leave this or open for a while.

File: enforcer-rules/pom.xml [hunk 1]
```diff
@@ -103,9 +103,11 @@
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-util</artifactId>
     </dependency>
+    <!-- needed for ArtifactCollector and maven-dependency-tree 2.2 -->
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-compat</artifactId>
+      <scope>compile</scope>
```
Comment (pzygielo) [inline enforcer-rules/pom.xml:110]: `scope` doesn't seem to be managed to anything else in parent, so why is it needed here and not for `maven-resolver-util` above?

**Comment by slachiewicz** [comment]
Thx merged with additional commits

**Comment by jira-importer** [comment]
Resolve #486

**Pull request closed without merging**